packages/ui-components/src/components/toaster/toaster.tsx

```diff
@@ -0,0 +1,72 @@
+import { Component, Event, Host, h, Prop, EventEmitter, State } from '@stencil/core';
+import { isEmpty } from 'lodash-es';
+import { EToasterType, EToasterIconTypeClass, IToaster, IToasterEvents, TYPE_ICONS, CLOSE_ICON } from './toaster.types';
+
+@Component({
+	tag: 'kv-toaster',
+	styleUrl: 'toaster.scss',
+	shadow: true
+})
+export class KvToaster implements IToaster, IToasterEvents {
+	/** @inheritdoc */
+	@Prop({ reflect: true }) header!: string;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) description?: string;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) type!: EToasterType;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) ttl?: number;
+	/** @inheritdoc */
+	@Event() clickCloseButton: EventEmitter<MouseEvent>;
+	/** State to store the timeout id */
+	@State() timeoutID: number;
+
+	private iconTypeClass: EToasterIconTypeClass = TYPE_ICONS[this.type];
+
+	private onCloseClick = event => {
+		window.clearTimeout(this.timeoutID);
+		this.closeToaster();
+		this.clickCloseButton.emit(event);
+	};
+
+	private closeToaster = () => {
+		if (this.ttl > 0) window.clearTimeout(this.timeoutID);
```

Comment:
Something is missing on this function in order to close the toaster. However, I have doubts about its implementation. Since we are using controllers for this component, should the close toaster due to ttl logic be implemented on the controller side or on the toaster side?

Reply:
Will do some experiments by adding an event to tell the parent component to close the toaster.